api/tests/unit/domain/validations/campaign-validator_test.js

```diff
@@ -309,69 +309,95 @@ describe('Unit | Domain | Validators | campaign-validator', function() {
     });
 
     context('on targetProfileId attribute', () => {
-      it('should reject with error when targetProfileId is provided and campaign type is PROFILES_COLLECTION', () => {
-        // given
-        const expectedError = {
-          attribute: 'targetProfileId',
-          message: 'Un profil cible n’est pas autorisé pour les campagnes de collecte de profils.'
-        };
 
-        try {
-          // when
-          campaignValidator.validate({
-            ...campaignOfTypeProfilesCollection,
-            targetProfileId: '1',
-          });
-          expect.fail('should have thrown an error');
+      context('when type is PROFILES_COLLECTION', () => {
+        it('should reject with error when targetProfileId is provided and campaign type is PROFILES_COLLECTION', () => {
+          // given
+          const expectedError = {
+            attribute: 'targetProfileId',
+            message: 'Un profil cible n’est pas autorisé pour les campagnes de collecte de profils.'
+          };
+
+          try {
+            // when
+            campaignValidator.validate({
+              ...campaignOfTypeProfilesCollection,
+              targetProfileId: '1',
+            });
+            expect.fail('should have thrown an error');
 
-        } catch (entityValidationErrors) {
-          // then
-          _assertErrorMatchesWithExpectedOne(entityValidationErrors, expectedError);
-        }
-      });
+          } catch (entityValidationErrors) {
+            // then
+            _assertErrorMatchesWithExpectedOne(entityValidationErrors, expectedError);
+          }
+        });
 
-      it('should reject with error when targetProfileId is missing and campaign type is ASSESSMENT', () => {
-        // given
-        const expectedError = {
-          attribute: 'targetProfileId',
-          message: 'Veuillez sélectionner un profil cible pour votre campagne.'
-        };
+        it('should be valid with null as targetProfileId and campaign type is PROFILES_COLLECTION', () => {
+          try {
+            campaignValidator.validate({
+              ...campaignOfTypeProfilesCollection,
+              targetProfileId: null,
+            });
+          } catch (entityValidationErrors) {
+            expect.fail('should be valid when targetProfileId is null');
+          }
+        });
 
-        try {
-          // when
-          campaignValidator.validate({
-            ...campaignOfTypeAssessment,
-            targetProfileId: MISSING_VALUE,
-          });
-          expect.fail('should have thrown an error');
+        it('should be valid when targetProfileId is not provided', () => {
+          try {
+            campaignValidator.validate({
+              ...campaignOfTypeProfilesCollection,
+            });
```

Comment:
```suggestion
            campaignValidator.validate({
              ...campaignOfTypeProfilesCollection,
              targetProfileId: UNDEFINED_VALUE,
            });
```
pour que ce soit clair

Reply:
👍